RoboJackets/igvc-software
# New Ground Segmentation Approach

# Description

This PR replaces our old RANSAC-based ground segmentation by using the approach defined [here](https://github.com/RoboJackets/igvc-software/blob/master/documents/research/barrel_detection.md). It introduces a new filter for the `pointcloud_filter` node, named `fast_segment_filter`. It publishes to the preexisting `/lidar/free` and `/lidar/occupied` topics as well as two new topics for debugging named `/ground` and `/nonground`.

This PR does the following:
- Creates new pointcloud filter to filter ground points from nonground points
- Contains configuration parameters in `igvc_perception/config/pointcloud_filter.yaml`

Fixes #473

# Testing steps (If relevant)
1. Run `roslaunch igvc_gazebo autonav.launch`
2. Run `roslaunch igvc_perception pointcloud_filter.launch`
3. Open `rviz`

Expectation: You should see nonground points like barrels in the `/nonground` topic and ground points in the `/ground` topic.


# Self Checklist
